assets/js/study.js

```diff
@@ -161,9 +175,30 @@ async function handle_move(orig, dest) {
     update_progress();
 }
 
-function create_shape(move, brush) {
-    let ft = san_to_uci(chess, move);
-    return {orig: ft.from, dest: ft.to, brush: brush};
+function array_contains(arr, str) {
```

Comment:
Already defined in ground.js. Probably best to refactor this out to some general util.js

Reply:
Done!